### PR TITLE
Update the forward ECal to leave the space for ECal insert

### PR DESCRIPTION
**Pull request by jizhongling**
### Briefly, what does this PR introduce?
Share the same source code "src/EndcapCalorimeterWithInsert_geo.cpp" with HCal.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No. The users need to use the updated forward ECal XML file.

### Does this PR change default behavior?
Yes, the default forward ECal XML file has been changed.

### CODE REVIEW

File: compact/ecal_forward_homogeneous.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<lccdd>
+
+  <info name="ecal_forward_homogeneous.xml"
+        title="Homogeneous forward ECal"
+        author="@rymilton"
+        url="https://github.com/rymilton"
```
Comment (wdconinc) [inline compact/ecal_forward_homogeneous.xml:6]: This should be changed into your username and github account.

File: compact/ecal_forward_homogeneous.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      />
+      <layer repeat="EcalEndcapPLayer_NRepeat" thickness = "EcalEndcapPSingleLayerThickness">
+        <slice name="Cover_slice" material="Aluminum" thickness="EcalEndcapPCoverThickness" vis="AnlRed"/>
+        <slice name="Air_slice" material="Air" thickness="EcalEndcapPAirThickness" vis="AnlRed"/>
+        <slice name="PCB_slice" material="Fr4" thickness="EcalEndcapPPCBThickness" vis="AnlRed"/>
+        <slice name="LightGuide_slice" material="Plexiglass" thickness="EcalEndcapPLightGuideThickness" vis="AnlBlue"/>
+        <slice name="Scintillator_slice" material="AvgTungstenScFi" thickness="EcalEndcapPScintillatorThickness" vis="AnlOrange" sensitive="true"/>
+      </layer>
+      <insert>
+        <position x="EcalEndcapPInsert_xposition" y="0" z = "0"/>
```
Comment (wdconinc) [inline compact/ecal_forward_homogeneous.xml:71]:
```suggestion
        <position x="EcalEndcapPInsert_xposition" y="0" z="0"/>
```

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: compact/ecal_forward_homogeneous.xml (hunk 1 above)
Comment (jizhongling) [reply compact/ecal_forward_homogeneous.xml]: I changed to my GitHub account.

File: compact/ecal_forward_homogeneous.xml (hunk 2 above)
Comment (jizhongling) [reply compact/ecal_forward_homogeneous.xml]: Thanks for finding this format typo. I removed the space.

**Comment by rymilton** [comment]
I initially asked @jizhongling (thank you!) to implement the ecal insert for me since I have little time with classes starting. I found some time though and am able to implement it myself. Will make a new pull request for just the ecal insert.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**